src/network/networkd-link.c

```diff
@@ -4368,6 +4371,18 @@ int link_save(Link *link) {
                         fputc('\n', f);
                 }
 
+                r = sd_dhcp_lease_get_client_id(link->dhcp_lease, &client_id, &client_id_len);
+                if (r >= 0) {
+                        _cleanup_free_ char *client_id_hex = NULL;
+
+                        client_id_hex = hexmem(client_id, client_id_len);
```

Comment:
Should this be `sd_dhcp_client_id_to_string()` instead of encoding the binary? Then, `sd_network_link_get_dhcp4_client_id()` becomes simpler.

Reply:
yeah good idea . Thanks.